6278 fixes broken tab download

landreev opened the pull request:
## New Contributors

This fixes the tab download issue introduced in #6244; there's also a minor cleanup of the API messages/return codes

## Related Issues

- closes #6278 

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

djbrooke commented:
@landreev, is this one ready for review?

djbrooke commented:
Thanks @pdurbin.

@landreev, my bad if I added this to the code review column prematurely. I went to do it, then I thought I'd ask instead, but it looks like I added it. Sorry!

landreev commented:
> Thanks @pdurbin.
> 
> @landreev, my bad if I added this to the code review column prematurely. I went to do it, then I thought I'd ask instead, but it looks like I added it. Sorry!

No, it was all good - I put it into code review myself yesterday. 
It's almost done at this point.
